frame/staking/src/pallet/mod.rs

```diff
@@ -1161,7 +1163,11 @@ pub mod pallet {
 			}
 
 			ensure!(!targets.is_empty(), Error::<T>::EmptyTargets);
-			ensure!(targets.len() <= T::MaxNominations::get() as usize, Error::<T>::TooManyTargets);
+			ensure!(
+				targets.len() <=
+					T::NominationsQuota::get_quota(Self::weight_of(&stash).into()) as usize,
```

Comment:
I don't see any reason to use `weight_of` here -- `get_quota` needs a balance, and we aleady have it at hand, namely via `ledger.active` (`weight_of` is reading this internally, can double check).

Reply:
`weight_of` is `VoteWeight`, so it's using `ledger.active`, but then `CurrencyToVote`'ing it.

Reply:
Changed to using `ledger.active` directly.